cudax/include/cuda/experimental/__device/device.cuh

```diff
@@ -48,6 +65,29 @@ public:
   {
     return __id;
   }
+
+  //! @brief Retrieve the specified attribute for the device
+  //!
+  //! @param attr The attribute to query. See `device::attrs` for the available
+  //!        attributes.
+  //!
+  //! @throws cuda_error if the attribute query fails
+  //!
+  //! @sa device::attrs
+  template <::cudaDeviceAttr Attr>
+  _CCCL_NODISCARD auto attr([[maybe_unused]] __attr<Attr> attr) const
+  {
+    int __value = 0;
+    _CCCL_TRY_CUDA_API(cudaDeviceGetAttribute, "failed to get device attribute", &__value, Attr, get());
```

Comment:
Lets try and always fully qualify the calls to cuda runtime because I am scared of ADL. Also should ou be passing `attr` in here? 
```suggestion
    _CCCL_TRY_CUDA_API(::cudaDeviceGetAttribute, "failed to get device attribute", &__value, Attr, get());
```